src/users/tests/test_api.py

```diff
@@ -3,19 +3,30 @@
 from django.urls import reverse
 
 
+@pytest.fixture
+def auth_user(django_user_model):
+    """創建一個用於認證的用戶，不影響測試邏輯"""
+    return django_user_model.objects.create(
+        email="test_auth@example.com",
+        speaker_name="Auth User",
+        verified=True,
+        is_active=True,
+    )
+
+
 @pytest.mark.django_db
-def test_user_list_with_role_filter_exact_match(api_client, django_user_model):
+def test_user_list_with_role_filter_exact_match(api_client, django_user_model, auth_user):
```

Comment:
should create another user without the specified role, ensuring the filtering logic really works